crates/uv/tests/tool_run.rs

```diff
@@ -835,3 +836,33 @@ fn tool_run_without_output() {
     warning: `uv tool run` is experimental and may change without warning
     "###);
 }
+
+#[test]
+fn warn_no_executables_found() {
+    let context = TestContext::new("3.12").with_filtered_exe_suffix();
+    let tool_dir = context.temp_dir.child("tools");
+    let bin_dir = context.temp_dir.child("bin");
+
+    uv_snapshot!(context.filters(), context.tool_run()
+        .arg("requests")
+        .env("UV_TOOL_DIR", tool_dir.as_os_str())
+        .env("XDG_BIN_HOME", bin_dir.as_os_str()), @r###"
+    success: false
+    exit_code: 1
+    ----- stdout -----
+    The executable `requests` was not found.
+
+    ----- stderr -----
+    warning: `uv tool run` is experimental and may change without warning
+    Resolved 5 packages in [TIME]
+    Prepared 5 packages in [TIME]
+    Installed 5 packages in [TIME]
+     + certifi==2024.2.2
+     + charset-normalizer==3.3.2
+     + idna==3.6
+     + requests==2.31.0
+     + urllib3==2.2.1
+    warning: An executable named `requests` is not provided by package `requests`.
+    warning: `requests` does not provide any executables.
```

Comment:
🤔  We should probably display the second warning _instead_ of the first warning if we can.

Reply:
Done